pinot-common/src/main/java/org/apache/pinot/common/response/broker/BrokerResponseNative.java

```diff
@@ -485,4 +488,15 @@ public Map<String, String> getTraceInfo() {
   public void setTraceInfo(Map<String, String> traceInfo) {
     _traceInfo = traceInfo;
   }
+
+  @Override
+  public void setTablesQueried(@NotNull Set<String> tablesQueried) {
+    _tablesQueried = tablesQueried;
+  }
+
+  @Override
+  @NotNull
```

Comment:
The convention in this project has been to only mark fields / methods / parameters that are nullable with `@Nullable` and to assume that non annotated elements are never nullable.

Reply:
I would say the convention is to mandatory mark nullable fields/methods/parameters and to _recommend_ to do not mark not nullable. The later is mainly to do avoid _annotatiomania_, but it is fine for me in situations we want to empathize something is not null.

Probably in this case that is not necessary.

Reply:
> annotatiomania

😆 

> but it is fine for me in situations we want to empathize something is not null.

Makes sense, although I just realized that we've been using `javax.annotation.Nonnull` (complement of `javax.annotation.Nullable`) and not `javax.validation.constraints.NotNull` (the one being introduced in this PR).

Reply:
> assume that non annotated elements are never nullable.
I haven't seen this to be true. I use this field in #14110. That has no annotation but checks for null.

The choice is between `@NotNull` or checks for `if (object == null) {...}`.

Reply:
> The choice is between @NotNull or checks for if (object == null) {...}

We currently don't have any mechanism to enforce nullability checks (like https://github.com/uber/NullAway) btw; the annotation is more like a hint to callers.

Reply:
> The choice is between @NotNull or checks for 

As Yash also said, that is not correct. It is the opposite. You can assume nullable things are annotated with `@Nullable`.

> We currently don't have any mechanism to enforce nullability checks (like https://github.com/uber/NullAway) btw; the annotation is more like a hint to callers.

I've created a PR in the past trying to use it. The main problem is that we are breaking the rules very often. https://github.com/apache/pinot/pull/13741.

Reply:
> annotatiomania

That is a term used by Lukas Eder, the creator of jOOQ, partially to mock about Hibernate. He even had a funny web page dedicated to that, but it seems he removed it long ago :(

Reply:
> As Yash also said, that is not correct.

Oh right - I meant if there was enforcement etc. Right now it is dev/IDE hint.